woocommerce-gutenberg-products-block.php

```diff
@@ -18,7 +18,72 @@
 
 defined( 'ABSPATH' ) || exit;
 
-if ( version_compare( PHP_VERSION, '5.6.0', '<' ) ) {
+define( 'WOOCOMMERCE_BLOCKS__MINIMUM_WP_VERSION', '5.0' );
+define( 'WOOCOMMERCE_BLOCKS__MINIMUM_PHP_VERSION', '5.6' );
+
+if ( version_compare( $GLOBALS['wp_version'], WOOCOMMERCE_BLOCKS__MINIMUM_WP_VERSION, '<' ) ) {
+	if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+		error_log( // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+			sprintf(
+				/* translators: Placeholders are numbers, versions of WordPress in use on the site, and required by WordPress. */
+				esc_html__( 'Your version of WordPress (%1$s) is lower than the version required by WooCommerce Blocks (%2$s). Please update WordPress to continue enjoying WooCommerce Blocks.', 'woo-gutenberg-products-block' ),
+				$GLOBALS['wp_version'],
+				WOOCOMMERCE_BLOCKS__MINIMUM_WP_VERSION
+			)
+		);
+	}
+	/**
+	 * Outputs for an admin notice about running WooCommerce Blocks on outdated WordPress.
+	 *
+	 * @since $VID:$
+	 */
+	function woocommerce_blocks_admin_unsupported_wp_notice() { ?>
+		<div class="notice notice-error is-dismissible">
+			<p><?php esc_html_e( 'WooCommerce Blocks requires a more recent version of WordPress and has been paused. Please update WordPress to continue enjoying WooCommerce Blocks.', 'woo-gutenberg-products-block' ); ?></p>
+		</div>
+		<?php
+	}
+	add_action( 'admin_notices', 'woocommerce_blocks_admin_unsupported_wp_notice' );
+	return;
+}
+
+if ( version_compare( PHP_VERSION, WOOCOMMERCE_BLOCKS__MINIMUM_PHP_VERSION, '<' ) ) {
+	if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+		error_log( // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+			sprintf(
+				/* translators: Placeholders are numbers, versions of PHP in use on the site, and required by WooCommerce Blocks. */
+				esc_html__( 'Your version of PHP (%1$s) is lower than the version required by WooCommerce Blocks (%2$s). Please update PHP to continue enjoying WooCommerce Blocks.', 'woo-gutenberg-products-block' ),
+				esc_html( phpversion() ),
+				WOOCOMMERCE_BLOCKS__MINIMUM_PHP_VERSION
+			)
+		);
+	}
+	/**
+	 * Outputs an admin notice for folks running an outdated version of PHP.
+	 *
+	 * @todo: Remove once WP 5.2 is the minimum version.
+	 *
+	 * @since $VID:$
+	 */
+	function woocommerce_blocks_admin_unsupported_php_notice() {
+		?>
+		<div class="notice notice-error is-dismissible">
+			<p><?php esc_html_e( 'WooCommerce Blocks requires a more recent version of PHP and has been paused. Please update PHP to continue enjoying WooCommerce Blocks.', 'woo-gutenberg-products-block' ); ?></p>
+			<p class="button-container">
+				<?php
+				printf(
+					'<a class="button button-primary" href="%1$s" target="_blank" rel="noopener noreferrer">%2$s <span class="screen-reader-text">%3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a>',
+					esc_url( wp_get_update_php_url() ),
```

Comment:
This function was added to WordPress in version 5.1, so it won't be available if the plugin is in a WordPress 5.0 environment and has something less than PHP 5.6.  So you can't use this function here if this notice stays.

Reply:
Good catch... I think we can remove that button completely. Done in 0ec623e032452e58d71dff16ced750695c948ebb.